js/components/tabs.jsx

```diff
@@ -0,0 +1,190 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+class Tabs extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedTab: 0,
+      tabControls: ['Request directions', 'Agency information'],
+    };
+    this.handleClickTab = this.handleClickTab.bind(this);
+    this.renderTabControls = this.renderTabControls.bind(this);
+    this.renderTabControl = this.renderTabControl.bind(this);
+  }
+
+  handleClickTab(index, event) {
+    event.preventDefault();
+    this.setState({
+      selectedTab: index,
+    });
+  }
+
+  renderTabControl(name, index) {
+    const key = `tab-control_${index}`;
+    const className = (index === this.state.selectedTab) ? 'tab_active' : '';
+    return (
+      <li
+        key={key}
+        className={className}
+      >
+        <a
+          role="tab"
+          tabIndex="0"
+          onClick={this.handleClickTab.bind(this, index)}
+        >
+          {name}
+        </a>
+      </li>
+    );
+  }
+
+  renderTabControls(props) {
+    const tabControls = this.state.tabControls.map((name, index) => (
+      this.renderTabControl(name, index)
+    ));
+    return (
+      tabControls
+    );
+  }
+
+  render() {
+    const tabControls = this.renderTabControls(this.props);
+    return (
+      <div className="sidebar">
+        <ul className="sidebar_tab-controls">
+          { this.renderTabControls() }
```

Comment:
You _could_ inline all of this...
```javascript
{ this.state.tabControls.map((name, index) => this.renderTabControl(name, index)) }
```

But maybe you prefer having the helper functions? The helper functions can make it easier to test...